homeassistant/components/aurora_abb_powerone/config_flow.py

```diff
@@ -56,55 +89,109 @@ def validate_and_connect(
     return ret
 
 
-def scan_comports() -> tuple[list[str] | None, str | None]:
-    """Find and store available com ports for the GUI dropdown."""
-    com_ports = serial.tools.list_ports.comports(include_links=True)
-    com_ports_list = []
-    for port in com_ports:
-        com_ports_list.append(port.device)
-        _LOGGER.debug("COM port option: %s", port.device)
-    if len(com_ports_list) > 0:
-        return com_ports_list, com_ports_list[0]
-    _LOGGER.warning("No com ports found.  Need a valid RS485 device to communicate")
-    return None, None
+def __validate_and_connect_tcp(hass: core.HomeAssistant, data):
+    host = data[CONF_HOST]
+    port = data[CONF_PORT]
+    address = data[CONF_ADDRESS]
+    _LOGGER.debug("Intitialising connection to %s:%s [%s]", host, port, address)
+    ret = {}
+    ret["title"] = DEFAULT_INTEGRATION_TITLE
+    try:
+        client = AuroraTCPClient(host, port, address, timeout=1)
+        client.connect()
+        ret[ATTR_SERIAL_NUMBER] = client.serial_number()
+        ret[ATTR_MODEL] = f"{client.version()} ({client.pn()})"
+        ret[ATTR_FIRMWARE] = client.firmware(1)
+        _LOGGER.info("Returning device info=%s", ret)
+    except AuroraError as err:
+        _LOGGER.warning("Could not connect to device %s:%s [%s]", host, port, address)
+        raise err
+    finally:
+        client.close()
+
+    # Return info we want to store in the config entry.
+    return ret
 
 
 class AuroraABBConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Aurora ABB PowerOne."""
 
     VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+    _comportslist = None
+    _defaultcomport = None
 
     def __init__(self):
         """Initialise the config flow."""
         self.config = None
-        self._com_ports_list = None
-        self._default_com_port = None
 
-    async def async_step_user(
-        self, user_input: dict[str, Any] | None = None
-    ) -> FlowResult:
+    async def async_step_user(self, user_input=None):
         """Handle a flow initialised by the user."""
+        return await self.async_step_user_ctype()
 
+    async def async_step_user_ctype(self, user_input=None):
+        """Select connection type."""
         errors = {}
-        if self._com_ports_list is None:
-            result = await self.hass.async_add_executor_job(scan_comports)
-            self._com_ports_list, self._default_com_port = result
-            if self._default_com_port is None:
-                return self.async_abort(reason="no_serial_ports")
+        if user_input is not None:
+            if user_input.get(CONF_TYPE) == CONF_TYPE_SERIAL:
+                if self._comportslist is None:
```

Comment:
This seems to have been reverted to a much older version that doesn't include the `async_add_executor_job`.  Please use the current code for scanning com ports.